File: src/main/java/xyz/mattyb/openwest/kotlinprimer/chapter05/Classes.java
package xyz.mattyb.openwest.kotlinprimer.chapter05;

public class Classes {
    public static void main(String[] args) {
        Object toCheck = "Checking";
        boolean isObject = toCheck instanceof Object;
        boolean isNumber = toCheck instanceof Number;
        System.out.println("String is object: " + isObject);
        System.out.println("String is number: " + isNumber);

        if (toCheck instanceof String) {
            String upper = ((String) toCheck).toUpperCase();
            System.out.println(upper);
        }

        String indented = ((String) toCheck).replaceFirst("C", "\tC");
        System.out.println(indented);
    }
}

File: src/main/java/xyz/mattyb/openwest/kotlinprimer/chapter05/classes.kt
package xyz.mattyb.openwest.kotlinprimer.chapter05

fun main(args: Array<String>) {
    /**
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     *
     * Don't want to give away how functions work ;-)
     */
    val toCheck: Any = "Checking"

    val isAny = toCheck is Any
    val isNumber = toCheck is Number

    println("String is any: $isAny, is number: $isNumber")

    if (toCheck is String) {
        val upper = toCheck.toUpperCase()

        println(upper)
    }

    val indented = (toCheck as String).prependIndent()
    println(indented)
}